diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index f1e58fe..7dadb48 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -1569,6 +1569,9 @@ void X11DRV_MotionNotify( HWND hwnd, XEvent *xev )
     TRACE( "hwnd %p/%lx pos %d,%d is_hint %d serial %lu\n",
            hwnd, event->window, event->x, event->y, event->is_hint, event->serial );
 
+    if(x11drv_thread_data()->xi2_state == xi_enabled)
+        return;
+
     input.u.mi.dx          = event->x;
     input.u.mi.dy          = event->y;
     input.u.mi.mouseData   = 0;
@@ -1669,16 +1672,6 @@ static void X11DRV_RawMotion( XGenericEventCookie *xev )
         break;
     }
 
-    if (thread_data->warp_serial)
-    {
-        if ((long)(xev->serial - thread_data->warp_serial) < 0)
-        {
-            TRACE( "pos %d,%d old serial %lu, ignoring\n", input.u.mi.dx, input.u.mi.dy, xev->serial );
-            return;
-        }
-        thread_data->warp_serial = 0;  /* we caught up now */
-    }
-
     TRACE( "pos %d,%d (event %f,%f)\n", input.u.mi.dx, input.u.mi.dy, dx, dy );
 
     input.type = INPUT_MOUSE;
